src/ray/gcs/gcs_server/job_info_handler_impl.cc

```diff
@@ -36,18 +40,23 @@ void DefaultJobInfoHandler::HandleAddJob(const rpc::AddJobRequest &request,
   if (!status.ok()) {
     on_done(status);
   }
-  RAY_LOG(DEBUG) << "Finished adding job, job id = " << job_id
-                 << ", driver pid = " << request.data().driver_pid();
 }
 
 void DefaultJobInfoHandler::HandleMarkJobFinished(
     const rpc::MarkJobFinishedRequest &request, rpc::MarkJobFinishedReply *reply,
     rpc::SendReplyCallback send_reply_callback) {
   JobID job_id = JobID::FromBinary(request.job_id());
   RAY_LOG(DEBUG) << "Marking job state, job id = " << job_id;
-  auto on_done = [job_id, reply, send_reply_callback](Status status) {
+  auto job_table_data =
+      gcs::CreateJobTableData(job_id, /*is_dead*/ true, std::time(nullptr), "", -1);
+  auto on_done = [this, job_id, job_table_data, reply,
+                  send_reply_callback](const Status &status) {
     if (!status.ok()) {
       RAY_LOG(ERROR) << "Failed to mark job state, job id = " << job_id;
+    } else {
+      RAY_CHECK_OK(gcs_pub_sub_->Publish(JOB_CHANNEL, job_id.Binary(),
```

Comment:
Will this lead to pub twice because that the `AsyncMarkFinished` will publish too?

Reply:
We also change the job subscribe code which only subscribe gcs_pub_sub_ publish messages.